spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -501,130 +503,83 @@ def __init__(self, parent=None):
-                               name='Yank', parent=self)
-        kill_ring_rotate = config_shortcut(self._kill_ring.rotate,
-                                           context='Editor',
-                                           name='Rotate kill ring',
-                                           parent=self)
-
-        kill_prev_word = config_shortcut(self.kill_prev_word, context='Editor',
-                                         name='Kill previous word',
-                                         parent=self)
-        kill_next_word = config_shortcut(self.kill_next_word, context='Editor',
-                                         name='Kill next word', parent=self)
-
-        start_doc = config_shortcut(cb_maker('Start'), context='Editor',
-                                    name='Start of Document', parent=self)
-
-        end_doc = config_shortcut(cb_maker('End'), context='Editor',
-                                  name='End of document', parent=self)
-
-        undo = config_shortcut(self.undo, context='Editor',
-                               name='undo', parent=self)
-        redo = config_shortcut(self.redo, context='Editor',
-                               name='redo', parent=self)
-        cut = config_shortcut(self.cut, context='Editor',
-                              name='cut', parent=self)
-        copy = config_shortcut(self.copy, context='Editor',
-                               name='copy', parent=self)
-        paste = config_shortcut(self.paste, context='Editor',
-                                name='paste', parent=self)
-        delete = config_shortcut(self.delete, context='Editor',
-                                 name='delete', parent=self)
-        select_all = config_shortcut(self.selectAll, context='Editor',
-                                     name='Select All', parent=self)
-        array_inline = config_shortcut(lambda: self.enter_array_inline(),
-                                       context='array_builder',
-                                       name='enter array inline', parent=self)
-        array_table = config_shortcut(lambda: self.enter_array_table(),
-                                      context='array_builder',
-                                      name='enter array table', parent=self)
-
-        return [codecomp, duplicate_line, copyline, deleteline, movelineup,
-                movelinedown, gotonewline, gotodef, toggle_comment, blockcomment,
-                unblockcomment, transform_uppercase, transform_lowercase,
-                line_start, line_end, prev_line, next_line,
-                prev_char, next_char, prev_word, next_word, kill_line_end,
-                kill_line_start, yank, kill_ring_rotate, kill_prev_word,
-                kill_next_word, start_doc, end_doc, undo, redo, cut, copy,
-                paste, delete, select_all, array_inline, array_table, indent,
-                unindent]
+        """Create the local shortcuts for the CodeEditor."""
+        return []

```

Comment:
All of the following widgets use the same architecture to handle shortcuts: `PythonShellWidget`,  `CodeEditor`, `EditorStack`, `ShellWidget`, and `FindReplace`. So I wanted to preserve this architecture in the `CodeEditor` as well, even if it will not be used after this PR. I do not have any good reason to do so, but removing this method would imply cleaning up things in several other places, and even in `mainwindow.py`. Maybe this should be done in another PR?

Reply:
I think it's fine to leave it for now. Please clarify why we don't use it in its docstring.